### PR TITLE
ISecurityStampValidator  /1

### LINKED ISSUE #26563: .NET 7 - Add documentation about Identity SecurityStampValidator
There's been an app building exercise around auth in 7.0, and that resulted in some feedback that we currently don't have any documentation about Identity's SecurityStamp / single sign out functionality.  I'll put the first draft of the content in this issue, but I wasn't sure exactly where this content should go, maybe a new link under authentication called Identity SIgnOut Everywhere?

# ISecurityStampValidator and SignOut everywhere

Sometimes your application does something that changes a user in such a way that its appropriate to regenerate a User's ClaimsPrincipal (joining a role, changing a password, or other security sensitive actions).  The `ISecurityStampValidator` is the interface that Identity uses to accomplish this.  The default implementation of Identity registers a SecurityStampValidator with the main application cookie, as well as the two factor cookie.  The validator hooks into the `OnValidatePrincipal` event of each cookie to call into Identity to verify that the user's SecurityStamp claim is unchanged from what's stored in the cookie, at regular intervals (tradeoff between hitting the database and stale claims). Whenever your app wants to force existing cookies to be invalid the next time they are checked, simply call `userManager.UpdateSecurityStampAsync(user)`, this is what most of the Identity UI account/manage pages do after changing the password/adding a login. This is also how apps can easily implement a SignOut everywhere action.

Changing the validation interval can be done via:
```C#
// Force Identity's security stamp to be validated every minute
builder.Services.Configure<SecurityStampValidatorOptions>(o => o.ValidationInterval = TimeSpan.FromMinutes(1));
```

cc @rafikiassumani-msft @blowdart @Tratcher @DamianEdwards @adityamandaleeka for feedback (should I elaborate more on the mechanics?)

---
[Associated WorkItem - 59735](https://dev.azure.com/msft-skilling/Content/_workitems/edit/59735)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #26563
[Internal review URL](https://review.learn.microsoft.com/en-us/aspnet/core/security/authentication/identity-configuration?view=aspnetcore-8.0&branch=pr-en-us-28125#isecuritystampvalidator-and-signout-everywhere)

## Public review  of the doc build:
- download the [HTML files](https://www.dropbox.com/sh/yy994551cje0kap/AAB0tRnm7gze2kYrSYPSFHIda?dl=0)
- unzip
- view HTML file in a browser.
- For links outside the doc, remove the `review.` from URL.

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@damienbod please review 
Hopefully I didn't change the meaning of #26563

**Comment by Rick-Anderson** [comment]
@damienbod  is this good to merge?

**Comment by damienbod** [comment]
reading now, one sec

File: aspnetcore/security/authentication/identity-configuration.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ uid: security/authentication/identity-configuration
 
 :::moniker range=">= aspnetcore-6.0"
 
+By [Damien Bowden](https://twitter.com/damien_bod)
+
```
Comment (damienbod) [inline aspnetcore/security/authentication/identity-configuration.md:16]: I did not write this, I should not get credit for this

**Comment by damienbod** [comment]
LGTM Greetings Damien

File: aspnetcore/security/authentication/identity-configuration.md (hunk 1 above)
Comment (Rick-Anderson) [reply aspnetcore/security/authentication/identity-configuration.md:16]: You reviewed it and I'm hoping you'll continue to review changes and consult. I can remove that line if you like until you've done more work on it.

File: aspnetcore/security/authentication/identity-configuration.md [hunk 2]
```diff
@@ -12,6 +12,8 @@ uid: security/authentication/identity-configuration
 
 :::moniker range=">= aspnetcore-6.0"
 
+By [Damien Bowden](https://twitter.com/damien_bod)
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/identity-configuration.md:15]:
```suggestion
```

File: aspnetcore/security/authentication/identity-configuration.md (hunk 1 above)
Comment (damienbod) [reply aspnetcore/security/authentication/identity-configuration.md:15]: @Rick-Anderson will do, thanks for asking

**Pull request merged**